### DISCUSSION TITLE
How to add FGM. In lighting I know how to add PGD against disturbance in torch module, but I don't know where to add FGM under lighting attack()

### DISCUSSION THREAD

**Discussion by Hou-jing** [Lightning-AI/pytorch-lightning#12680] [code help: NLP / ASR / TTS]
![image](https://user-images.githubusercontent.com/75936962/162597742-f9e31dad-053a-4980-96b9-66b2018cf5b0.png)

How to add FGM. In lighting I know how to add PGD against disturbance in torch module, but I don't know where to add FGM under lighting attack()

**Comment by akihironitta** [comment]
Hi @Hou-jing!

I think you can use manual optimization for your use-case. Check out the doc page for details: https://pytorch-lightning.readthedocs.io/en/latest/common/optimization.html#manual-optimization

**Reply by Hou-jing** [reply]
Thank you. I've solved my problem,
But I think the official should give some complete examples, which may be more convenient to use

**Reply by akihironitta** [reply]
It'll be fantastic if you could provide the example for the community! Any contribution is welcome as part of #10239.
